src/back/glsl/mod.rs

```diff
@@ -2467,27 +2476,22 @@ impl<'a, W: Write> Writer<'a, W> {
                 let right_inner = ctx.info[right].ty.inner_with(&self.module.types);
 
                 let function = match (left_inner, right_inner) {
-                    (
-                        &Ti::Vector {
-                            kind: left_kind, ..
-                        },
-                        &Ti::Vector {
-                            kind: right_kind, ..
-                        },
-                    ) => match op {
+                    (&Ti::Vector { kind, .. }, &Ti::Vector { .. }) => match op {
                         Bo::Less
                         | Bo::LessEqual
                         | Bo::Greater
                         | Bo::GreaterEqual
                         | Bo::Equal
                         | Bo::NotEqual => BinaryOperation::VectorCompare,
-                        Bo::Modulo => match (left_kind, right_kind) {
-                            (Sk::Float, _) | (_, Sk::Float) => match op {
-                                Bo::Modulo => BinaryOperation::Modulo,
-                                _ => BinaryOperation::Other,
-                            },
-                            _ => BinaryOperation::Other,
-                        },
+                        Bo::Modulo if kind == Sk::Float => BinaryOperation::Modulo,
+                        Bo::And if kind == Sk::Bool => {
+                            op = crate::BinaryOperator::LogicalAnd;
+                            BinaryOperation::VectorComponentWise
+                        }
```

Comment:
You can just do `vec & vec` why is this needed?

Reply:
It doesn't work for boolean vectors.

Reply:
Oh right this is for booleans my bad